spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -875,38 +930,43 @@ def process_signatures(self, params):
 
                 signature = signature_data['label']
                 parameter = parameter_data['label']
-                parameter_documentation = parameter_data['documentation']
+                # parameter_documentation = parameter_data['documentation']
 
                 # This method is part of spyder/widgets/mixins
                 self.show_calltip(
-                    signature,
-                    documentation,
-                    parameter,
-                    parameter_documentation,
-                    color='#999999',
-                    is_python=self.is_python()
+                    signature=signature,
+                    parameter=parameter,
                 )
         except Exception:
             self.log_lsp_handle_errors("Error when processing signature")
 
     # ------------- LSP: Hover ---------------------------------------
     @request(method=LSPRequestTypes.DOCUMENT_HOVER)
-    def request_hover(self, line, col):
+    def request_hover(self, line, col, show_hint=True):
         """Request hover information."""
         params = {
             'file': self.filename,
             'line': line,
             'column': col
         }
+        self._show_hint = show_hint
         return params
 
     @handles(LSPRequestTypes.DOCUMENT_HOVER)
     def handle_hover_response(self, contents):
         """Handle hover response."""
         try:
-            text = contents['params']
-            self.sig_display_signature.emit(text)
-            self.show_tooltip(_("Hint"), text)
+            content = contents['params']
+            self.sig_display_signature.emit(content)
+
+            if self._enable_hover_hints and content:

```

Comment:
According to my comment above, we should change this to

    if CONF.get('lsp-server', 'show_hints') and content: